src/main/java/lotto/domain/LottoNumber.java

```diff
@@ -6,11 +6,15 @@ public class LottoNumber {
 
     private static final int NUMBER_MIN = 1;
     private static final int NUMBER_MAX = 45;
-
     private final int number;
 
-    public LottoNumber(int number) {
+
+    public static LottoNumber from(int number) {
         checkBoundaryNumbers(number);
+        return LottoNumberCache.getNumber(number);
+    }
+
+    LottoNumber(int number) {
         this.number = number;
```

Comment:
유효성 검증은 생성자에 배치하는 것이 좋습니다.
- 기본 접근제한자여도 같은 패키지라면 범위를 벗어난 LottoNumber는 생성될 수 있습니다.
- 항상 from 메소드를 통해 생성한다는 보장도 없고, 
새로운 정적 팩토리 메소드가 생성되면 유효성 검증 로직이 또 들어가게됩니다. 🥲

이렇기 때문에 검증 로직은 생성자에 들어가는 것이 올바른 방법 같아요~
```suggestion

    LottoNumber(int number) {
        checkBoundaryNumbers(number);
        this.number = number;
    }
    
    public static LottoNumber from(int number) {
        return LottoNumberCache.getNumber(number);
    }
```

그리고 **정적 팩토리 메소드도 메소드**이기 때문에 컨벤션에 맞게 생성자 아래에 배치하면 좋겠습니다.